prog/dftb+/lib_dftbplus/initprogram.F90

```diff
@@ -4085,44 +4064,53 @@ end subroutine checkTransportRanges
 
 
   !> Clean up things that did not automatically get removed by going out of scope
-  subroutine destructProgramVariables()
+  subroutine destructProgramVariables(this)
 
-    if (electronicSolver%isElsiSolver) then
-      call TElsiSolver_final(electronicSolver%elsi)
+    !> Instance
+    class(TDftbPlusMain), intent(inout) :: this
+
+    if (this%electronicSolver%isElsiSolver) then
+      call TElsiSolver_final(this%electronicSolver%elsi)
     end if
 
-    if (tProjEigenvecs) then
-      call destruct(iOrbRegion)
-      call destruct(RegionLabels)
+    if (this%tProjEigenvecs) then
+      call destruct(this%iOrbRegion)
+      call destruct(this%regionLabels)
     end if
 
-    @:SAFE_DEALLOC(sccCalc, img2CentCell, species, species0, coord, coord0)
-    @:SAFE_DEALLOC(latVec, origin, recVec, invLatVec, cellVec, rCellVec, iCellVec)
-    @:SAFE_DEALLOC(neighbourList, nNeighbourSk, nNeighbourRep, iSparseStart)
-    @:SAFE_DEALLOC(hubbU, atomEigVal, referenceN0, mass, speciesMass)
-    @:SAFE_DEALLOC(ham, iHam, chargePerShell, chargePerAtom, over, kPoint, kWeight)
-    @:SAFE_DEALLOC(nEl, spinW, xi, UJ, nUJ, niUJ, iUJ, Ef, esp)
-    @:SAFE_DEALLOC(indMovedAtom, conAtom, conVec, pipekMezey)
+    @:SAFE_DEALLOC(this%sccCalc, this%img2CentCell, this%species, this%species0, this%coord)
```

Comment:
why not deallocate (this) instead of its components?

Reply:
You're absolutely right, it came from 'mechanical transformation'. I've changed the logic now.